juju/model.py

```diff
@@ -1005,16 +1005,26 @@ def debug_log(
                 # haven't made it yet we'll need to wait on them to be added
                 await asyncio.gather(*[
                     asyncio.ensure_future(
-                        self._wait_for_new('application', app_name))
+                        self._wait_for_new('application', app_name),
+                        loop=self.loop)
                     for app_name in pending_apps
-                ])
+                ], loop=self.loop)
             return [app for name, app in self.applications.items()
                     if name in handler.applications]
         else:
-            log.debug(
-                'Deploying %s', entity_id)
-
             if not is_local:
+                parts = entity_id[3:].split('/')
```

Comment:
Just verified that a local charm path doesn't have to start with `./` (at least, absolute paths work fine)